framework/src/outputs/XDA.C

```diff
@@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
 void
 XDA::output(const ExecFlagType & /*type*/)
 {
+  std::string fname = filename();
+  std::string base = fname.substr(0, fname.size()-4);
   if (_binary)
   {
-    _mesh_ptr->getMesh().write(filename()+"_mesh.xdr");
-    _es_ptr->write (filename()+".xdr", ENCODE, EquationSystems::WRITE_DATA | EquationSystems::WRITE_ADDITIONAL_DATA);
+    _mesh_ptr->getMesh().write(base + "_mesh.xdr");
```

Comment:
@permcody The filename() method is used by the OutputWarehouse to check if multiple files exist with the same name. Hence, it really needs the extension to be accurate. Currently, for xda/xdr it doesn't.

The XDR/XDA object does use filename, it is the name used by the EquationSystems object write (`_es_ptr->write`). This object also needs to write a second file for the mesh, thus the "_mesh" suffix is inserted.